### PR TITLE
Resolves ReadTimeout bug of #26

### LINKED ISSUE #26: [Bug] => Unhandled Timeout Exception
#### Describe the bug
There is a bug where the timeout exception is not being handled properly. An unresponsive site with default timeout 7s gives this error.

#### Command You Used
`python3 xsrfprobe.py -u http://xxxxxxxxx.xx/csrf --display --malicious --max-chars 4`

#### Full Stack Trace Error
```
Traceback (most recent call last):
  File "C:\Python37\lib\site-packages\urllib3\connectionpool.py", line 384, in _make_request
    six.raise_from(e, None)
  File "<string>", line 2, in raise_from
  File "C:\Python37\lib\site-packages\urllib3\connectionpool.py", line 380, in _make_request
    httplib_response = conn.getresponse()
  File "C:\Python37\lib\http\client.py", line 1321, in getresponse
    response.begin()
  File "C:\Python37\lib\http\client.py", line 296, in begin
    version, status, reason = self._read_status()
  File "C:\Python37\lib\http\client.py", line 257, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "C:\Python37\lib\socket.py", line 589, in readinto
    return self._sock.recv_into(b)
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python37\lib\site-packages\requests\adapters.py", line 449, in send
    timeout=timeout
  File "C:\Python37\lib\site-packages\urllib3\connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "C:\Python37\lib\site-packages\urllib3\util\retry.py", line 367, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "C:\Python37\lib\site-packages\urllib3\packages\six.py", line 686, in reraise
    raise value
  File "C:\Python37\lib\site-packages\urllib3\connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "C:\Python37\lib\site-packages\urllib3\connectionpool.py", line 386, in _make_request
    self._raise_timeout(err=e, url=url, timeout_value=read_timeout)
  File "C:\Python37\lib\site-packages\urllib3\connectionpool.py", line 306, in _raise_timeout
    raise ReadTimeoutError(self, url, "Read timed out. (read timeout=%s)" % timeout_value)
urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='www.hemsheela.in', port=80): Read timed out. (read timeout=7)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\user\Videos\xsrfprobe\core\request.py", line 79, in Get
    req = requests.get(url, headers=headers, timeout=TIMEOUT_VALUE, stream=False)
  File "C:\Python37\lib\site-packages\requests\api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "C:\Python37\lib\site-packages\requests\api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\Python37\lib\site-packages\requests\sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Python37\lib\site-packages\requests\sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "C:\Python37\lib\site-packages\requests\adapters.py", line 529, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPConnectionPool(host='www.hemsheela.in', port=80): Read timed out. (read timeout=7)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "xsrfprobe.py", line 13, in <module>
    main.Engine()  # start the Scanner Engine ;)
  File "C:\Users\user\Videos\xsrfprobe\core\main.py", line 192, in Engine
    soup = crawler.process(fld)  # Start the parser
  File "C:\Users\user\Videos\xsrfprobe\modules\Crawler.py", line 65, in process
    query = Get(url)  # Open it (to check if it exists)
  File "C:\Users\user\Videos\xsrfprobe\core\request.py", line 100, in Get
    ErrorLogger(main_url, e.__str__())
NameError: name 'main_url' is not defined
```
#### Potential cause or fix
Proper handling of the `requests.exceptions.ReadTimeout` along with proper naming of variable `main_url` will solve it.

#### Environment:
- OS: Windows 10
- Python version: v3.7

#### Some Questions

- [x] I am using the latest version of XSFRProbe.
- [x] I installed the dependecies using `pip3` instead of `pip`.
- [x] I have read the documentation before submitting this issue.
- [x] I have checked the other issues to see if someone reported this before.

#### Other stuff
I'll make a PR for this. :)

### PR DESCRIPTION
**Pull request by iDuronto**
#### Which category is this pull request?

- [ ] A new feature/enhancement.
- [x] Fix an issue/feature-request.
- [x] An improvement to existing modules.
- [ ] Other (Please mention below).

#### Where has this been tested?
- Python Version: v3.7
- Operating System: Windows 10

#### Does this close any currently open issues? 
#26 

#### Does this add any new dependency?
Neh.

#### Does this add any new command line switch/option?
Nah.

#### Any other comments you would like to make?
This PR resolves #26.

#### Some Questions

- [x] I have documented my code.
- [x] My code follows the PEP 8 style guide.
- [x] I have tested my build before submitting the pull request.

### CODE REVIEW

**Comment by iDuronto** [comment]
Build passed. Ready for your review.

File: core/request.py [hunk 1]
```diff
@@ -87,15 +92,20 @@ def Get(url, headers=headers):
...
+    except requests.exceptions.ReadTimeout as e:
+        verbout(R, 'Exception at: '+color.GREY+url)
+        verbout(R, 'Error: Read Timeout. Consider increasing the timeout value via --timeout.')
+        ErrorLogger(url, e.__str__())
+        return None
     except requests.exceptions.HTTPError as e:  # if error
-        verbout(R, "HTTP Error : "+main_url)
-        ErrorLogger(main_url, e.__str__())
+        verbout(R, "HTTP Error Encountered : "+main_url)
+        ErrorLogger(url, e.__str__())
```
Comment (0xInfection) [inline core/request.py:102]: Ah this was a big mistake which I managed to overlook. :(

**Review by 0xInfection — APPROVED** [review]
Yeah, everything's pretty neat and according to default code infrastructure.

File: core/request.py (hunk 1 above)
Comment (iDuronto) [reply core/request.py:102]: Yeah, I fixed it. Always there for you, bruh.

**Pull request merged**